### PR TITLE
Better error propagation

### PR DESCRIPTION
**Pull request by bakulf**

### CODE REVIEW

File: src/errorhandler.h [hunk 1]
```diff
@@ -48,6 +48,14 @@ class ErrorHandler final : public QObject {
   };
   Q_ENUM(AlertType)
 
+  // This can be an boolean but it's better to have an enum to improve the
+  // readability of the code.
+  enum ErrorPropagation {
+    NoErrorPropagation,
+    PropagateError,
+  };
+  Q_ENUM(ErrorPropagation);
```
Comment (brizental) [inline src/errorhandler.h:57]: It's weird to me that this enum is defined on the error handler when in fact its being used elsewhere in the code. AFAICT it's only used in the Tasks, what about we move this declaration there?

File: src/errorhandler.h [hunk 2]
```diff
@@ -48,6 +48,14 @@ class ErrorHandler final : public QObject {
   };
   Q_ENUM(AlertType)
 
+  // This can be an boolean but it's better to have an enum to improve the
+  // readability of the code.
+  enum ErrorPropagation {
+    NoErrorPropagation,
+    PropagateError,
```
Comment (brizental) [inline src/errorhandler.h:55]: This is kinda nitpicky, but how about `DoNotPropagateError` / `PropagateError`?

File: src/tasks/account/taskaccount.cpp [hunk 3]
```diff
@@ -25,8 +25,15 @@ void TaskAccount::run() {
...
       request, &NetworkRequest::requestFailed, this,
       [this](QNetworkReply::NetworkError error, const QByteArray&) {
-        logger.error() << "Account request failed" << error;
-        ErrorHandler::instance()->errorHandle(ErrorHandler::toErrorType(error));
+        logger.error() << "Account request failed" << error
+                       << "error-propagation:"
+                       << (m_errorPropagation == ErrorHandler::PropagateError
+                               ? "yes"
+                               : "no");
+        if (m_errorPropagation == ErrorHandler::PropagateError) {
```
Comment (brizental) [inline src/tasks/account/taskaccount.cpp:33]: I am not a big fan of re-declaring this conditional in each task, it makes it easier for the meaning or "ErrorPropagation" to be changed in sutil ways as new tasks are implemented or changed. What if we declare a function on the Task super class like so:

```cpp
void handleTaskError(Error error) {
   if (m_errorPropagation == ErrorHandler::PropagateError) {
       ErrorHandler::instance()->errorHandle(
              ErrorHandler::toErrorType(error));
   }
}
```

**Review by brizental — CHANGES_REQUESTED** [review]

File: src/releasemonitor.cpp [hunk 4]
```diff
@@ -1,72 +1,75 @@
...
-    connect(task, &Task::completed, [] {
-      MozillaVPN* vpn = MozillaVPN::instance();
-      Q_ASSERT(vpn);
-      vpn->setUpdating(false);
-    });
-
-    TaskScheduler::scheduleTask(task);
-  });
-}
+/* This Source Code Form is subject to the terms of the Mozilla Public
```
[the remark this answers is not in the document]
Comment (bakulf) [reply src/releasemonitor.cpp:1]: because the file was encoded using CR+LF (dos/windows)...

File: src/errorhandler.h (hunk 1 above)
Comment (bakulf) [reply src/errorhandler.h:57]: I thought about that, but why tasks? we can have the same enum in every single class or implement it centralized. ErrorHandler seems to be the right choice. Balrog is not a task and still need the ErrorHandler enum

Comment (brizental) [reply src/errorhandler.h]: Ah! I missed Balrog. Ok yeah, that makes it tricker.

File: src/errorhandler.cpp [hunk 5]
```diff
@@ -226,3 +226,12 @@ void ErrorHandler::setAlert(AlertType alert) {
   m_alert = alert;
   emit alertChanged();
 }
+
+// static
+void ErrorHandler::networkErrorHandle(
```
Comment (brizental) [inline src/errorhandler.cpp:231]: nit: `handleNetworkError`? `networkErrorHandle` sounds like the function is going to return a handle to something.

File: src/errorhandler.h (hunk 1 above)
Comment (brizental) [reply src/errorhandler.h]: Ah, the new commit changes thing in a way that it makes complete sense for the enum to be on the ErrorHandler class.

**Review by brizental — APPROVED** [review]
r+wc

File: src/errorhandler.cpp (hunk 5 above)
Comment (brizental) [reply src/errorhandler.cpp:231]: Oh I see that `errorHandle` also has "handle" at the end. IMO it also sounds like that function returns a handle to something... But we don't need to make this PR about renaming a bunch of things.

**Comment by dlmchugh1212** [comment]
please stop sending

**Pull request merged**